### PR TITLE
Integrate Contentful

### LINKED ISSUE #44: Migrate to Contentful
## Description

Migrate static text content to Contentful.

### PR DESCRIPTION
**Pull request by zorazrr**
## Summary

Currently, `home.json` is providing text data to home page, we are shifting to Contentful, which will allow more flexibility for editing and prepare us for a potential blog feature. 

To access all contents, login to [Contentful](https://app.contentful.com/spaces/gslvrj5c21wx/entries?id=L7Zf1hc9FwOY66ng&order.fieldId=updatedAt&order.direction=descending&displayedFieldIds=contentType&displayedFieldIds=updatedAt&displayedFieldIds=author&folderId=default&page=0) with the credentials shared on Notion. 

Closes #44 

## Changes

- Add packages `contentful` and `@contentful/rich-text-react-renderer`
- Set up `ContentWrapper` to initialize contentful client 
- Use contentful client on `home/index.tsx` 

## Content Types 

```ts
export interface OurRequestsEntry {
  order: number
  title: string
  details: Document
}

export interface LinkEntry {
  name: string
  url: string
}

export interface ListEntry {
  order: number
  details: string
}

export interface RefuteResponseEntry {
  order: number
  details: Document
}

export interface TimelineEntry {
  year: number
  details: string
}

export interface Info {
  landing: Document
  refuteUisResponse: Document
}
``` 
image

## Additional Notes 

- Changes in `lint.yml` workflow are necessary to prevent [this error](https://github.com/toriis-portal/toriis/actions/runs/4422851139/jobs/7755029074)
- `LinkDisplay` is moved to the `InstitutionalDivestment` section file to maintain structural consistency and prevent prop-drilling

### CODE REVIEW

File: src/info/home.json
Comment (daniel-moon32) [inline src/info/home.json:1]: Let's gooo no more json file

File: src/sections/home/Landing.tsx [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
   CapitalizedTitle,
 } from '../../components'
 
-const Landing: FC = () => {
-  const textData = HomeData.landingContent
-  const landingTextStyle = clsx(
-    'font-semibold underline decoration-clementine decoration-2 underline-offset-4',
-  )
+const Landing: FC<{ text: Document }> = ({ text }) => {
+  const contentfulOptions = {
```
Comment (daniel-moon32) [inline src/sections/home/Landing.tsx:14]: More of the whacky contentful stuff. I think I get it, but for future reference you probably want a quick comment to tell people what this does.

**Review by daniel-moon32 — CHANGES_REQUESTED** [review]
Just  a couple small things with naming, but otherwise looks good

File: src/info/home.json
Comment (zorazrr) [reply src/info/home.json:1]: Let's gooo commenting on the entire file

File: src/pages/home/index.tsx [hunk 2]
```diff
@@ -8,8 +8,53 @@ import {
...
+  OurRequestsEntry,
+  RefuteResponseEntry,
+  TimelineEntry,
+} from '../../types'
 
-const Home: FC = () => {
+export const getServerSideProps = async () => {
+  const contentClient = new ContentWrapper()
+  const info = await contentClient.get('info')
+  const ourRequestsEntries = await contentClient.get('request')
```
[the remark this answers is not in the document]
Comment (zorazrr) [reply src/pages/home/index.tsx]: Seems that Contentful does not support get entries by multiple content types. I made it slightly better by adding a `getAllHomePageEntries` function in the newest commit. It's still making 6 requests though lol. Another option is to call `getEntries` without content type to get everything, but we might want to add content types in the future so not doing that might be the way to go.

**Review by daniel-moon32 — APPROVED** [review]
Looks good!

**Pull request merged**